datadog_checks_dev/datadog_checks/dev/config_validator/config_block.py

```diff
@@ -0,0 +1,452 @@
+        if param_prop.type_name.startswith('list'):
+            end = _get_end_of_param_declaration_block(
+                start, len(config_lines), config_lines, indent, errors, is_list=True
+            )
+            if end is None:
+                default_end = _get_next_block_in_case_of_failure(start, config_lines)
+                return ConfigBlock(None, None, start, default_end - start, errors)
+            block_len = end - start
+
+        # Parsing the description
+        idx += 1
+        description, idx = _parse_description(idx, end, config_lines, indent, errors)
+        if idx is None:
+            return ConfigBlock(param_prop, None, start, block_len, errors)
+
+        # We recurse if the variable is an object and contains at least one member with description
+        is_object = _is_object(idx, config_lines, indent, param_prop, errors)
+        if not is_object:
+            return ConfigBlock(param_prop, description, start, block_len, errors, should_recurse=False)
+
+        should_recurse, next_block = _should_recurse(idx, config_lines, indent)
+        if should_recurse:
+            # If we recurse we use block_len, pointing to the next sub-block
+            return ConfigBlock(param_prop, description, start, block_len, errors, should_recurse=True)
+
+        # If we don't recurse we use the next_object variable to point to the next block with the same or less
+        # indentation and thus ignore sub-blocks.
+        block_len = next_block - start
+        return ConfigBlock(param_prop, description, start, block_len, errors, should_recurse=False)
+
+
+def _get_end_of_param_declaration_block(start, end, config_lines, indent, errors=None, is_list=False):
+    """Here we suppose the config block is correctly formatted (@param, description, empty comment then the actual content)
+    and try to return the line of any data coming after. In case of a object we point to its first member. In case
+    of a list or a simple variable we point to the next element.
+    """
+    if errors is None:
+        errors = []
+
+    if not is_exactly_indented(config_lines[start], indent):
+        other_indent = get_indent(config_lines[start])
+        errors.append(ValidatorError("Unexpected indentation, expecting %d not %d" % (indent, other_indent), start))
+        return
+
+    if not config_lines[start].startswith(' ' * indent + "## @param"):
+        errors.append(ValidatorError("Expecting @param declaration", start))
+        return
+
+    # Going through the description
+    idx = start + 1
+    while idx < end:
+        if is_blank(config_lines[idx]):
+            errors.append(ValidatorError("Blank line when reading description", idx))
+            return
+        if not is_exactly_indented(config_lines[idx], indent):
+            other_indent = get_indent(config_lines[idx])
+            errors.append(ValidatorError("Unexpected indentation, expecting %d not %d" % (indent, other_indent), idx))
+            return None
```

Comment:
It's a bit inconsistent with above doing `return` and here `return None`.

Reply:
Adding consistency and choosing `return None` instead of `return`